rasa/core/brokers/pika.py

```diff
@@ -371,343 +123,213 @@ def _get_queues_from_args(
-        ),
-        **kwargs: Any,
+    @staticmethod
+    async def _bind_queue(
+        queue_name: Text, channel: aio_pika.RobustChannel, exchange: aio_pika.Exchange
     ) -> None:
-        """Initialise RabbitMQ event broker.
-
-        Args:
-            host: Pika host.
-            username: Username for authentication with Pika host.
-            password: Password for authentication with Pika host.
-            port: port of the Pika host.
-            queues: Pika queues to declare and publish to.
-            should_keep_unpublished_messages: Whether or not the event broker should
-                maintain a queue of unpublished messages to be published later in
-                case of errors.
-            raise_on_failure: Whether to raise an exception if publishing fails. If
-                `False`, keep retrying.
-            log_level: Logging level.
-        """
-        logging.getLogger("pika").setLevel(log_level)
-
-        self.host = host
-        self.username = username
-        self.password = password
-        self.port = port
-        self.queues = queues
-        self.process: Optional[multiprocessing.Process] = None
-        self.should_keep_unpublished_messages = should_keep_unpublished_messages
-        self.raise_on_failure = raise_on_failure
-        self.pika_message_processor: Optional[PikaMessageProcessor] = None
-
-        self._connect()
+        queue = await channel.declare_queue(queue_name, durable=True)
+        await queue.bind(exchange, "")
 
     def __del__(self) -> None:
-        if self.pika_message_processor:
-            self.pika_message_processor.close()
-
-        if self.process and self.process.is_alive():
-            self.process.terminate()
+        """Closes connection when object is destroyed."""
+        self._loop.run_until_complete(self._close())
 
     def close(self) -> None:
-        """Close the Pika connector."""
+        """Close the pika channel and connection."""
         self.__del__()
 
-    @classmethod
-    def from_endpoint_config(
-        cls, broker_config: Optional["EndpointConfig"]
-    ) -> Optional["PikaEventBroker"]:
-        """Initialise `PikaEventBroker` from `EndpointConfig`.
+    async def _close(self) -> None:
+        if not self._connection:
+            return
 
-        Args:
-            broker_config: `EndpointConfig` to read.
+        # Entering the context manager does nothing. Exiting closes the channels and
+        # the connection.
+        async with self._connection:
+            logger.debug("Closing RabbitMQ connection.")
 
-        Returns:
-            `PikaEventBroker` if `broker_config` was supplied, else `None`.
-        """
-        if broker_config is None:
-            return None
-
-        return cls(broker_config.url, **broker_config.kwargs)
+    def is_ready(self) -> bool:
+        """Return `True` if a connection was established."""
+        return self._exchange is not None
```

Comment:
What do you think of actually checking if the connection is ready here? Or at least updating the docstring to make it less confusing why we check for `self._exchange` to be `not None`.

Reply:
adapted it